src/fontlibc/fontlibc.h

```diff
@@ -0,0 +1,437 @@
+/**
+ * @file
+ * @authors DrDnar
+ * @brief Provides improved font support
+ */
+
+#ifndef H_FONTLIBC
+#define H_FONTLIBC
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define FONTLIB_ENABLE_AUTO_WRAP 1
+#define FONTLIB_AUTO_CLEAR_TO_EOL 2
+#define FONTLIB_PRECLEAR_NEWLINE 4
```

Comment:
These should ideally be enum values.

Reply:
enums are ints, so then I'd either have to make the setter take an int when I only allocated a byte, or make users typecast. Which is more evil?

Reply:
Enum constants are automatically narrowed to the target type. The signatures of the functions accepting these values would remain unchanged. But you could add doc to specify that the flags should correspond to this group of flags that now has a handy concrete name.